### PR TITLE
Ms.networking2

### PR DESCRIPTION
**Pull request by mariano54**
Cleaning up websocket server and RPCs.
Also speeds up the tests.
Fix LGTM errors
Fix 0 connections display in wallet
Fix full node height 0 after switching keys
Fix introducer vetting
Outbound messages now go out asynchronously as opposed to in order
Refactor setup nodes to use Service
Log rotation with concurrency (not tested on windows yet)
Fix double logging

### CODE REVIEW

File: src/server/start_service.py [hunk 1]
```diff
@@ -199,7 +200,7 @@ def stop(self):
     async def wait_closed(self):
         await self._task
         if self._rpc_task:
-            await self._rpc_task
+            await (await self._rpc_task)()
```
[the remark this answers is not in the document]
Comment (mariano54) [reply src/server/start_service.py:203]: I think it was bugged, it was running the method, but that method returns a method (cleanup) which wasn't being called

File: src/rpc/rpc_server.py [hunk 2]
```diff
@@ -145,6 +142,9 @@ async def ws_api(self, message):
             return pong()
 
         f = getattr(self, command, None)
+        if f is not None:
```
[the remark this answers is not in the document]
Comment (mariano54) [reply src/rpc/rpc_server.py:145]: OK I suppose we can have a separate class for the connection APIs

File: src/wallet/wallet_node.py [hunk 3]
```diff
@@ -77,20 +77,21 @@ class WalletNode:
...
         root_path: Path,
         name: str = None,
         override_constants: Dict = {},
         local_test: bool = False,
     ):
-        self = WalletNode()
         self.config = config
+        if local_test:
+            self.config["database_path"] = "test_db_wallet.db"
+
```
[the remark this answers is not in the document]
Comment (mariano54) [reply src/wallet/wallet_node.py:94]: Yeah this is not new code (it was in websocket server) but should be fixable

File: src/server/connection.py [hunk 4]
```diff
@@ -174,7 +177,7 @@ def add(self, peer: Optional[PeerInfo]) -> bool:
         if peer is None or not peer.port:
             return False
         if peer not in self._peers:
-            self._peers.append(peer)
+            self._peers.append(PeerInfo(peer.host, peer.port))
```
Comment (richardkiss) [inline src/server/connection.py:180]: ? Why? Maybe needs a comment

Comment (mariano54) [reply src/server/connection.py:180]: Oh I was trying to debug memory leaks. I think this is a useless change

**Pull request merged**